helix-term/src/ui/menu.rs

```diff
@@ -231,8 +231,10 @@ impl<T: Item + 'static> Component for Menu<T> {
             key!(Enter) => {
                 if let Some(selection) = self.selection() {
                     (self.callback_fn)(cx.editor, Some(selection), MenuEvent::Validate);
+                    return close_fn;
+                } else {
+                    return EventResult::Ignored(None);
```

Comment:
This will keep the prompt open, you want to use the callback from close_fn (but with ignored instead of consumed)

Reply:
Actually this does close it - I think it might be because of the auto-closing behavior also introduced in #1285. I pushed a change with it explicitly popping the compositor to get rid of the menu though. Seems like good behavior to explicitly clean that up :+1: